fix: constrain version of jax

### Description

The CI pipeline is failing with the trace:
```
ImportError while loading conftest '/home/runner/work/jumanji/jumanji/jumanji/conftest.py'.
jumanji/__init__.py:19: in <module>
    from jumanji.env import Environment
jumanji/env.py:20: in <module>
    import chex
/opt/hostedtoolcache/Python/3.9.[16](https://github.com/instadeepai/jumanji/actions/runs/5142210251/jobs/9255958037?pr=125#step:7:17)/x64/lib/python3.9/site-packages/chex/__init__.py:[17](https://github.com/instadeepai/jumanji/actions/runs/5142210251/jobs/9255958037?pr=125#step:7:18): in <module>
    from chex._src.asserts import assert_axis_dimension
/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/chex/_src/asserts.py:26: in <module>
    from chex._src import asserts_internal as _ai
/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/chex/_src/asserts_internal.py:34: in <module>
    from chex._src import pytypes
/opt/hostedtoolcache/Python/3.9.16/x64/lib/python3.9/site-packages/chex/_src/pytypes.py:32: in <module>
    ArrayDevice = jax.interpreters.xla.DeviceArray
E   AttributeError: module 'jax.interpreters.xla' has no attribute 'DeviceArray'
```

### What Jumanji version are you using?

_No response_

### Which accelerator(s) are you using?

_No response_

### Additional System Info

_No response_

### Additional Context

_No response_

### (Optional) Suggestion

The error trace from the CI pipeliune indicates that the version of JAX being used is using the new `jax.Array` type, possibly with the `DeviceArray` deprecated. Hence, we most put an upper bound on the version of JAX being installed.

--- a/requirements/requirements.txt
+++ b/requirements/requirements.txt
@@ -1,7 +1,7 @@
 chex>=0.1.3,<0.1.6
 dm-env>=1.5
 gym>=0.22.0
-jax>=0.2.26
+jax>=0.2.26,<=0.4.1
 jaxlib>=0.1.74
 matplotlib>=3.3.4
 numpy>=1.19.5

--- a/requirements/requirements.txt
+++ b/requirements/requirements.txt
@@ -2,7 +2,7 @@ chex>=0.1.3,<0.1.6
 dm-env>=1.5
 gym>=0.22.0
 jax>=0.2.26,<=0.4.1
-jaxlib<=0.4.1
+jaxlib>=0.1.74,<=0.4.1
 matplotlib>=3.3.4
 numpy>=1.19.5
 Pillow>=9.0.0

--- a/requirements/requirements.txt
+++ b/requirements/requirements.txt
@@ -2,6 +2,7 @@ chex>=0.1.3,<0.1.6
 dm-env>=1.5
 gym>=0.22.0
 jax>=0.2.26,<=0.4.1
+jaxlib<=0.4.1
 matplotlib>=3.3.4
 numpy>=1.19.5
 Pillow>=9.0.0

--- a/requirements/requirements.txt
+++ b/requirements/requirements.txt
@@ -2,7 +2,6 @@ chex>=0.1.3,<0.1.6
 dm-env>=1.5
 gym>=0.22.0
 jax>=0.2.26,<=0.4.1
-jaxlib>=0.1.74
 matplotlib>=3.3.4
 numpy>=1.19.5
 Pillow>=9.0.0

